src/ophyd_async/tango/signal/_signal.py

```diff
@@ -0,0 +1,215 @@
+
+    Parameters
+    ----------
+    datatype:
+        Check that the Attribute/Command is of this type
+    write_trl:
+        The Attribute/Command to write to
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the write operation
+    name:
+        The name of the Signal
+    """
+    if datatype is None:
+        datatype = infer_python_type(write_trl)
+    backend = make_backend(datatype, write_trl, write_trl, device_proxy)
+    return SignalW(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_x(
+    write_trl: str,
+    device_proxy: Optional[DeviceProxy] = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> SignalX:
+    """Create a `SignalX` backed by 1 Tango Attribute/Command
+
+    Parameters
+    ----------
+    write_trl:
+        The Attribute/Command to write its initial value to on execute
+    device_proxy:
+        If given, this DeviceProxy will be used
+    timeout:
+        The timeout for the command operation
+    name:
+        The name of the Signal
+    """
+    backend = make_backend(None, write_trl, write_trl, device_proxy)
+    return SignalX(backend, timeout=timeout, name=name)
+
+
+# --------------------------------------------------------------------
+def tango_signal_auto(
+    datatype: Optional[Type[T]] = None,
+    *,
+    trl: str,
+    device_proxy: Optional[DeviceProxy] = None,
+    timeout: float = DEFAULT_TIMEOUT,
+    name: str = "",
+) -> Union[SignalW, SignalX, SignalR, SignalRW]:
+    if datatype is None:
+        datatype = infer_python_type(trl)
+    backend = make_backend(datatype, trl, trl, device_proxy)
+    signal = infer_signal_frontend(trl, name, timeout)
+    signal._backend = backend  # noqa: SLF001
+
+    return signal
+
+
+# --------------------------------------------------------------------
+def infer_python_type(trl: str) -> Type[T]:
+    device_trl, tr_name = trl.rsplit("/", 1)
+    syn_proxy = SyncDeviceProxy(device_trl)
+
+    if tr_name in syn_proxy.get_command_list():
+        config = syn_proxy.get_command_config(tr_name)
+        isarray, py_type, _ = get_python_type(config.in_type)
+    elif tr_name in syn_proxy.get_attribute_list():
+        config = syn_proxy.get_attribute_config(tr_name)
+        isarray, py_type, _ = get_python_type(config.data_type)
+        if py_type is Enum:
+            enum_dict = {label: i for i, label in enumerate(config.enum_labels)}
+            py_type = IntEnum("TangoEnum", enum_dict)
+        if config.data_format in [AttrDataFormat.SPECTRUM, AttrDataFormat.IMAGE]:
+            isarray = True
+    else:
+        raise RuntimeError(f"Cannot find {tr_name} in {device_trl}")
+
+    if py_type is CmdArgType.DevState:
+        py_type = DevState
+
+    return npt.NDArray[py_type] if isarray else py_type
+
+
+# --------------------------------------------------------------------
```

Comment:
This is very nitpicky but I'd prefer these dividers were removed for consistency with the rest of the codebase

Reply:
Done.